src/sync.cpp

```diff
@@ -113,7 +103,7 @@ static void potential_deadlock_detected(const std::pair<void*, void*>& mismatch,
 
 static void push_lock(void* c, const CLockLocation& locklocation)
 {
-    LockData& lockdata = GetLockData();
+    LockData& lockdata{*g_lockdata.Get()};
     std::lock_guard<std::mutex> lock(lockdata.dd_mutex);
```

Comment:
Another alternative to local `lockdata` is to use global `g_lockdata`:
```cpp
std::lock_guard<std::mutex> lock(g_lockdata->dd_mutex);
```
Would result in a bigger diff but I think it is better than "aliases".